python/cudf/cudf/core/column/numerical.py

```diff
@@ -718,6 +735,8 @@ def _normalize_find_and_replace_input(
     if isinstance(col_to_normalize, list):
         if normalized_column.null_count == len(normalized_column):
             normalized_column = normalized_column.astype(input_column_dtype)
+        if normalized_column.can_cast_safely(input_column_dtype):
```

Comment:
I guess without this change and all the ones below it we would wind up just raising before, and those are some of the pytest.raises that are removed?

Reply:
Yup, we were raising incorrectly and this change will help us not to raise unnecessarily.